docs/UserGuide.md

```diff
@@ -153,14 +176,67 @@ Examples:
 * `edittask 1 n/read chapter 5 p/HIGH` edits the first task name to `read chapter 5` and
 and priority to `HIGH`.
 
-#### Finding a task: `findtask`
-
-Finds a task based on keywords.
-
-Format: `findtask` **_`[KEYWORD]`_**
+#### Locating tasks: `findtask`
+
+Finds all tasks that fulfil all the provided search criteria.
+
+Format: `findtask [n/NAME_KEYWORDS] [d/DATE] [p/PRIORITY] [t/TAG_KEYWORDS]` 
+
+ * The search is case-insensitive, e.g. `lab` will match `Lab`.
+ 
+ * When you are providing name or tag keywords, separate distinct keywords with a whitespace, 
+   e.g. `findtask n/lab quiz` will search for tasks using the 2 distinct keywords `lab` and `quiz`.
+   
+ * You should ensure that search arguments are not blank and at least one search parameter should be provided.
+ 
+ * Search Parameters:
+ 
+   * Task Name
+ 
+     * You are allowed to provide multiple search keywords.
+     
+     * Tasks with their name matching at least one of the name keywords provided will be considered to have fulfilled the task name search criteria.
+     
+     * The order of the search keywords does not matter, e.g. `Lab Quiz` will match `Quiz Lab`.
+     
+     * Only full words will be matched, e.g. `lab` will match `lab assignment` but not `labs`.
+     
+   * Date
+   
+     * Your search date should be of the format: `YYYY-MM-DD`
+     
+     * You should only provide one search date
+     
+     * Tasks with their date matching the search date exactly are considered to have fulfilled the task date search criteria.
+     
+   * Priority
+   
+     * Your search priority should be one of the following: `highest`, `high`, `normal`, `low` (case-insensitive). 
+       No other search priority will be allowed.
+       
+     * You should only provide one search priority.
```

Comment:
May I know if this means the user cannot input `findtask p/high highest`? Maybe we can add examples of incorrect commands to make things clearer?

Reply:
Yes, the above mentioned input would not be allowed.  Yes, I would add in clearer examples. Thanks for the note 👍